Analytics datastore refactoring

aaemnnosttv opened the pull request:
## Summary

Addresses issue #1269

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

aaemnnosttv commented:
@felixarntz @tofumatt the only thing missing here seems to be the CLA consent from you two so I'm going to move this to review.

tofumatt commented:
@googlebot I consent.

felixarntz commented on stories/module-analytics-settings.stories.js:
```diff
@@ -0,0 +1,210 @@
+	);
+}
+
+storiesOf( 'Analytics Module Settings', module )
```
Let's change this to `Analytics Module/Settings`. This will make `Settings` be a separate "sub-story" of `Analytics Module`, which makes sense here.

aaemnnosttv replied:
I didn't realize stories could be organized like this. Awesome! 👍

felixarntz commented on stories/module-analytics-setup.stories.js:
```diff
@@ -0,0 +1,176 @@
+	);
+}
+
+storiesOf( 'Analytics Module Setup', module )
```
Same here, let's use `Analytics Module/Setup`.

aaemnnosttv commented:
@felixarntz I made a few updates here that were missed before. Mostly storybook updates, but a few tweaks to components, etc:

* Added the same `useEffect` we use in `SettingsEdit` to `SetupMain` for enforcing the existing tag values. This didn't get updated when we changed from doing this in the datastore to in the component (we only updated `SettingsEdit`).
* I moved the `ErrorNotice` component from both "main" components and into the inner view components. The reason for this is that a few views (loading/progress, and `ExistingTagError`) we don't want to render the error (particularly in the `ExistingTagError` case as it would show two errors on the screen). This was a fairly simple change.
* One update to a tag permission selector which was still using both property and account IDs but also because we have a dedicated selector for that.
* Updated the `INITIAL_STATE.siteInfo` to be `undefined` (rather than an object of `undefined` values for consistency, but also for simplifying the check in the resolver)
* Added the early return in `resolvers.getSiteInfo` if `siteInfo` was already set. This was causing errors in one of the component tests even though the data was being received in the test setup, because the resolver was not returning early like we do in others. This is done in a single commit so we can cherry pick it for another PR, etc if we need it in develop sooner than this PR will be (although I hope that's not the case 😄 )

Storybook fixes were things like removing old actions like `applyProperty` that didn't exist anymore, updating `receiveExistingTag` to get the string propertyID and a few other minor changes, but these should all be working now.

aaemnnosttv commented:
Regarding that last commit, I had to make one more change to the `core/site` store related to the `siteInfo` change from before where it was always returning an object which is inconsistent with our other selectors. I updated it to return `undefined` if the data isn't loaded yet and updated the sub-selectors to use `select(...) || {}` to prevent errors while preserving the behavior there. I talked to @tofumatt about this before making this change since he did the original datastore work there. The intention was just to prevent the specific-selectors from failing. Tests are all good now! ✅
